src/tablist.js

```diff
@@ -226,6 +226,29 @@ SideTabList.prototype = {
         browser.tabs.remove(tabId);
       }
     });
+    items.push({
+      label: browser.i18n.getMessage("contextMenuUndoCloseTab"),
+      onCommandFn: () => {
+        function restoreMostRecent (sessionInfos) {
+          if (!sessionInfos.length) {
+            console.log("No sessions found");
+            return;
+          }
+          let sessionInfo = sessionInfos[0];
+          if (sessionInfo.tab) {
+            browser.sessions.restore(sessionInfo.tab.sessionId);
+          }
+        }
+
+        function onError (error) {
+          console.log(error);
+        }
+
+        browser.sessions.getRecentlyClosed({
```

Comment:
So you mean always display the menu item but enable it asynchronously ?
I'll check that this evening.

Reply:
Yep, sounds about right. There's not CSS class to show that an element is in a disabled state, so you might have to add that.

Reply:
Hum I'm kind of lost.

After some tests, I can assure that `onCommandFn` will only be called with a click on the menu item. So `getRecentlyClosed` is only called when you click on the menu item.

For me, this is not related to when the menu is displayed or not.

So maybe, you were talking to add this behavior only to disable the menu item if there was no tab closed yet?

But the problem is that if I add an async function to check if the menu item should be enabled or not, I'll have to call `getRecentlyClosed` to check if there is a tab already closed, and this time yes, this will add a delay to display the menu.. :/

And apparently, there is no way to get the current session without calling `getRecentlyClosed` so I'm not sure if this is even possible to do this.

Reply:
But maybe, we could disable the menu by default and if the event `tabs.onRemoved` is fired, enable it. Not sure if it's feasible or not. I'll try.

Reply:
What I was saying is that isEnabled should be a promise-returning function.
In this function you would check for getRecentlyClosed().
And in contextmenu.js you would check for isEnabled (and hide if necessary) asynchronously after rendering the elements first.
You're not too far away from the finish line by the way 👍

Reply:
Well I think I get it this time. :)

I tried to hide the menu instead of greying it but I didn't like the result as when displaying the menu, you see the menu item then it disappears.. So I prefer to keep it greyed and disable the event when it's clicked.